storage/innobase/xtrabackup/src/xtrabackup.cc

```diff
@@ -2156,11 +2156,103 @@ static bool innodb_init_param(void) {
   return (TRUE);
 }
 
+/** Scan the sdi id from mysql.tables
+@param[in]	name 		Name of table database/name
+@param[out]     sdi_id		id of table
+@param[in]	thd 		THD */
+void get_id_from_mysql_table_scan(std::string name, uint64 *id, THD *thd) {
+  std::string db_name = name.substr(0, name.find(OS_PATH_SEPARATOR));
+
+  std::string table_name = name.substr(name.find(OS_PATH_SEPARATOR) + 1);
+  if (table_name.find("#P#") != std::string::npos ||
+      table_name.find("#p#") != std::string::npos) {
+    table_name.resize(table_name.find('#'));
+  }
+  uint64 schema_id = 0;
+
+  ut_ad(db_name.compare("mysql") != 0);
+
+  dict_table_t *sys_tables = nullptr;
+  btr_pcur_t pcur;
+  const rec_t *rec = nullptr;
+  mtr_t mtr;
+  MDL_ticket *mdl = nullptr;
+  mem_heap_t *heap = mem_heap_create(1000);
+
+  mutex_enter(&(dict_sys->mutex));
+
+  auto build_mysql_schema = [&]() {
+    std::map<std::string, int> mysql_schema;
+
+    mtr_start(&mtr);
+    rec = dd_startscan_system(thd, &mdl, &pcur, &mtr, "mysql/schemata",
+                              &sys_tables);
+    while (rec) {
+      MDL_ticket *mdl_on_tab = nullptr;
+      uint64 rec_schema_id;
+      std::string rec_name;
+
+      dd_process_schema_rec(heap, rec, sys_tables, &mdl_on_tab, &mtr, &rec_name,
+                            &rec_schema_id);
+      mysql_schema.insert(std::pair<std::string, int>(rec_name, rec_schema_id));
+      mem_heap_empty(heap);
+
+      mtr_start(&mtr);
+      rec = (rec_t *)dd_getnext_system_rec(&pcur, &mtr);
+    }
+
+    mtr_commit(&mtr);
+    dd_table_close(sys_tables, thd, &mdl, true);
+    mem_heap_empty(heap);
+
+    return mysql_schema;
+  };
+
+  static auto mysql_schema = build_mysql_schema();
+
+  schema_id = mysql_schema[db_name];
+  ut_a(schema_id != 0);
+
+  mtr_start(&mtr);
+
+  rec =
+      dd_startscan_system(thd, &mdl, &pcur, &mtr, "mysql/tables", &sys_tables);
+
+  while (rec) {
+    MDL_ticket *mdl_on_tab = nullptr;
+    uint64 rec_schema_id;
+    std::string rec_name;
+    uint64 rec_id;
+
+    dd_process_dd_tables_rec(heap, rec, sys_tables, &mdl_on_tab, &mtr,
+                             &rec_schema_id, &rec_name, &rec_id);
+    mem_heap_empty(heap);
+
+    if (schema_id == rec_schema_id && table_name.compare(rec_name) == 0) {
```

Comment:
@rahulmalik87 Can you also please test with a table name that has non-ascii characters?

Reply:
it doesn't work, we need to handle it.  thanks you for pointing

Reply:
please add such table in zipped backup dir.

Reply:
added